### ISSUE TITLE
[RFC] Switch to JavaScript

### ISSUE THREAD

**Issue by umpirsky** [umpirsky/centipede#27]
I am thinking to switch this project to JavaScript.

I am considering [PhantomJS](http://phantomjs.org/) and simple plugin system (similar to https://github.com/umpirsky/centipede/issues/26) with [Cucumber.js](https://github.com/cucumber/cucumber-js).

**Comment by akovalyov** [comment]
My two cents.

I am not sure if PhantomJS is good enough for that job, it sometimes gives false-negative due to internal failures, so probably CasperJS/ZombieJS can do the job well here.  
But my question is - why should we tie to a specific browser/engine? Why not https://www.npmjs.com/package/selenium-webdriver or any other JS browser driver?

**Comment by umpirsky** [comment]
Good point. If possible we can use selenium webdriver. I have also seen other js drivers which can be handy for this purpose.

I just want to figure something that will make plugins like https://github.com/umpirsky/centipede/issues/26 easy to implement (without reinventing the wheel), since this was the biggest blocker with PHP implementation.

**Comment by umpirsky** [comment]
I just had a closer look at both. CasperJS looks like exactly what we need, and is easier to work and create plugins with.
